ENIGMAsystem/SHELL/Platforms/General/POSIX/main.cpp

```diff
@@ -23,8 +24,13 @@ void initialize_directory_globals() {
 
   // Set the program_directory
   buffer[0] = 0;
-  ssize_t count = readlink("/proc/self/exe", buffer, PATH_MAX + 1);
-  if (count !=  -1)
+  char path[PATH_MAX + 1];
+  memset(buffer, 0, sizeof(buffer));
+  
+  pid_t pid = getpid();
+  sprintf(path, "/proc/%d/exe", pid);
```

Comment:
This is a lot of stack space you're consuming to do two string concatenations. I'd recommend just `("/proc/" + std::tostring(getpid()) + "/exe").c_str()`.